samcli/local/docker/lambda_build_container.py

```diff
@@ -46,6 +47,7 @@ def __init__(  # pylint: disable=too-many-locals
         source_dir = str(pathlib.Path(source_dir).resolve())
 
         container_dirs = LambdaBuildContainer._get_container_dirs(source_dir, manifest_dir)
+        env_vars = env_vars or {}
```

Comment:
nit: haven't seen this pattern much in Python, I'd just use a plain old `if`

Reply:
ah, I can change this to env_vars if env_vars else {}, but this pattern is used in our code in a couple of places for similar assignment, that's why I used this

Reply:
Or simply

```
if not env_vars:
    env_vars = {}
```

(which is also symmetrical to the pattern used in the file below)

Reply:
Actually I assume in these case the clearest would likely to default to an empty `dict`; would also get rid of the `Optional`.

Reply:
our pylint doesn't allow the default value to be an empty dict unfortunately, so I changed it to env_vars if env_vars else {}, but without removing the Optional